pkg/i2gw/providers/openapi3/storage.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openapi3
+
+import (
+	"sync"
+
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+type Storage interface {
+	AddResource(resource *openapi3.T)
+	GetResources() []*openapi3.T
+	Clear()
+}
+
+// NewResourceStorage returns a thread-safe storage for OpenAPI specs.
+func NewResourceStorage() Storage {
+	return &storage{}
+}
+
+type storage struct {
+	mu sync.RWMutex
```

Comment:
Why do we need thread safety here? I haven't seen any concurrency in the code, did I miss something?

Reply:
I tried to answer it here: https://github.com/kubernetes-sigs/ingress2gateway/pull/157#discussion_r1590105256

TL;DR - I think not having it is a road to problems in the future.

Not supposed to sound unnecessarily ominous here... I do believe things change, such as – I'm making stuff up here, ok? – start supporting multiple input files (or an input directory equivalently) and, if so, why not calling `ReadResourcesFromFile` concurrently?

It's a small price to pay IMO, but if more people believe we're better off _without_ the mutex, I can remove it – even if I don't understand why.